src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -682,3 +682,53 @@ ReplicateAllReferenceTablesToNode(WorkerNode *workerNode)
 		}
 	}
 }
+
+
+/*
+ * NodeHasAllReferenceTablesReplicas returns whether the given worker node has reference
+ * table replicas. If there are no reference tables the function returns true.
+ *
+ * This function does not do any locking, so the situation could change immediately after,
+ * though we can only ever transition from false to true, so only "false" could be the
+ * incorrect answer.
+ *
+ * In the case where the function returns true because no reference tables exist
+ * on the node, a reference table could be created immediately after. However, the
+ * creation logic guarantees that this reference table will be created on all the
+ * nodes, so our answer was correct.
+ */
+bool
+NodeHasAllReferenceTableReplicas(WorkerNode *workerNode)
+{
+	List *referenceTableIdList = CitusTableTypeIdList(REFERENCE_TABLE);
+
+	if (list_length(referenceTableIdList) == 0)
+	{
+		/* no reference tables exist */
+		return true;
+	}
+
+	Oid referenceTableId = linitial_oid(referenceTableIdList);
+	List *shardIntervalList = LoadShardIntervalList(referenceTableId);
```

Comment:
We seem to have a similar function, `WorkersWithoutReferenceTablePlacement`, can we maybe use that? 

Not critical ofc, just used needed something similar recently in #6173 and resembled this one

Reply:
I looked at it, but did not want to implement the join between WorkersWithoutReferenceTablePlacement and nodeIdsForPlacementList

Reply:
I looked at it, but it seems to do a lot of stuff I don't really need here.